src/components/DaySelect/calander_component/calander_component.tsx

```diff
@@ -88,14 +89,58 @@ export const CalanderComponent = ({
       return a.getTime() - b.getTime();
     });
 
+    setLastSelectedDate(date);
     setSelectedDates(arr);
   };
 
   const removeDay = (date: Date) => {
     const arr = selectedDates.filter((obj) => obj.getTime() != date.getTime());
+    setLastSelectedDate(date);
     setSelectedDates(arr);
   };
 
+  const handleRange = (date: Date) => {
```

Comment:
reorder the handleRange and getRange functions. It makes sense for handleRange to appear later in the code since it is used after a range is made. Makes it easier to scan the code and get a sense for what's happening.